docs/xxshareorphans.md

```diff
@@ -0,0 +1,65 @@
+|-----------------------|---------|:----------------------------------------------------------------------------------:|
+| `-XX:+ShareOrphans` |  Enable   |                                                                                    |
+| `-XX:-ShareOrphans` |  Disable  |   :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span>   |
+
+## Explanation
+
+In the previous versions, OpenJ9 stored only bootstrap classes and hidden classes in the shared classes cache by default, that is `-Xshareclasses` is not specified by default. When `-Xshareclasses` was specified, only those class loaders that implemented the OpenJ9's public shared classes cache APIs (and its child class loaders) could store classes to the shared classes cache. For classes from custom class loaders that did not implement shared classes cache APIs, the VM does not have their module or class path information. Such classes were not stored to the shared classes cache.
+
+You can enable class sharing from all class loaders, irrespective of whether the class loader implements the OpenJ9's public shared classes cache API, with the `-XX:+ShareOrphans` option. This option automatically turns on `-Xshareclasses`. When the class sharing from all class loaders is enabled, following is the sharing behavior:
+
+- For classes from class loaders that implement the OpenJ9 shared class API, they are shared as normal ROM classes, which is same as turning on `-Xshareclasses`.
+
+- For classes from class loaders that do not implement the OpenJ9 shared class API, the VM won't have their class or module path information. They are shared as orphan ROM classes with other VMs after extra comparisons.
+
+Storing additional classes in the cache makes more classes available for Ahead-of-Time (AOT) compilation and therefore might improve startup performance.
+
+You can disable sharing class as orphans from class loader that does not implement the OpenJ9 shared class API with the `-XX:-ShareOrphans` option. This option is the default mode.
+
+:fontawesome-solid-triangle-exclamation:{: .warn aria-hidden="true"} **Restrictions:**
+
+- Sharing classes as orphans requires more comparison on the classes by the VM. The comparison itself has a negative performance impact. However, the benefits of more AOT generated for the cached class might offset the negative impact.
+- More cached classes usually result in more AOT-compiled methods. The relative advantage of orphan sharing decreases when the CPUs are less. With more CPUs, the compilation threads can compile those additional methods in parallel with the application threads, while with fewer CPUs, extra compilation activity hinders application threads.
+- The class comparison might not detect the removal of method access modifiers. For example, a change of a method from public to package-private.
+- `java.lang.StackTraceElement.getClassLoaderName()` might return null for classes that are stored in the shared cache.
```

Comment:
> The class comparison might not detect the removal of method access modifiers. For example, a change of a method from public to package-private.

> `java.lang.StackTraceElement.getClassLoaderName()` might return null for classes that are stored in the shared cache.

I believe these 2 need to be included in the known problems list as well. Are you planning to mention them in `known problems` ?

Reply:
In the Release notes, yes.

Reply:
Please refer to the format we are using in the Release notes - https://github.com/eclipse-openj9/openj9/blob/master/doc/release-notes/0.45/0.45.md#known-issues

Could you please provide content that I can add in the "Impact" and "Workaround" columns for these 2 known issues.

Thanks!

Reply:
- The class comparison might not detect the removal of method access modifiers.
  Impact: If you recompiled classes removing method access modifier and rerun with `-XX:+ShareOrphans`, the old version of the class might still be used.
  Workaround: Remove `-XX:+ShareOrphans` and rerun.

- `java.lang.StackTraceElement.getClassLoaderName()` might return null for classes that are stored in the shared cache.
   Impact: `java.lang.StackTraceElement.getClassLoaderName()` returns null on class loader that actually has a name if running with `-XX:+ShareOrphans`.
   Workaround: Remove `-XX:+ShareOrphans` and rerun.

Reply:
In this sentence, "java.lang.StackTraceElement.getClassLoaderName() returns null on class loader that actually has a name if running with -XX:+ShareOrphans", what does "... if running with -XX:+ShareOrphans" relate to? I mean 
- is it "java.lang.StackTraceElement.getClassLoaderName() returns null if running with -XX:ShareOrphans..." or 
- is it "java.lang.StackTraceElement.getClassLoaderName() returns null on class loaders that are running with the -XX:+ShareOrphans option and actually have a name. "?

Reply:
> is it "java.lang.StackTraceElement.getClassLoaderName() returns null on class loaders that are running with the -XX:+ShareOrphans option and actually have a name. "?

This is more accurate in describing the problem. 
So the class loader may or may not have a name. If the class loader does not have a name, return null is correct. If class loader has a name, the correct behaviour is to return the name, but option `-XX:+ShareOrphans` could make the API return null incorrectly. I have edit the sentence a bit in https://github.com/eclipse-openj9/openj9-docs/pull/1383#discussion_r1697191385

Reply:
I have updated the Release notes as well with the known issues. https://github.com/eclipse-openj9/openj9/pull/19944

Please check and confirm. Thanks!